src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice/GasPriceOracle.cs

```diff
@@ -87,10 +114,10 @@ IEnumerable<Block> GetBlocks(long currentBlockNumber)
                 }
             }
             
-            return GetGasPricesFromRecentBlocks(GetBlocks(blockNumber), BlockLimit);
+            return GetGasPricesFromRecentBlocks(GetBlocks(blockNumber), numberOfBlocks, func);
         }
-        
-        private IEnumerable<UInt256> GetGasPricesFromRecentBlocks(IEnumerable<Block> blocks, int blocksToGoBack)
+
+        private IEnumerable<UInt256> GetGasPricesFromRecentBlocks(IEnumerable<Block> blocks, int blocksToGoBack, Func<Transaction, bool, UInt256, UInt256> func)
```

Comment:
maybe call it more specific than func? Also maybe lets create private delegate type with proper naming for parameters?